src/utils/git.rs

```diff
@@ -18,13 +18,27 @@ pub struct GitContext {
 
 impl GitContext {
     pub fn try_from_rover_env(env: &RoverEnv) -> Result<Self> {
-        let repo = Repository::discover(env::current_dir()?)?;
-        let head = repo.head().ok();
-
-        let branch = GitContext::get_branch(env, head.as_ref())?;
-        let commit = GitContext::get_commit(env, head.as_ref())?;
-        let committer = GitContext::get_committer(env, head.as_ref())?;
-        let remote_url = GitContext::get_remote_url(env, &repo)?;
+        let repo = Repository::discover(env::current_dir()?);
+        let branch;
+        let commit;
+        let committer;
+        let remote_url;
+
+        match repo {
+            Ok(repo) => {
+                let head = repo.head().ok();
+                branch = GitContext::get_branch(env, head.as_ref())?;
+                commit = GitContext::get_commit(env, head.as_ref())?;
+                committer = GitContext::get_committer(env, head.as_ref())?;
+                remote_url = GitContext::get_remote_url(env, Some(&repo))?;
+            }
+            Err(_) => {
+                branch = GitContext::get_branch(env, None)?;
+                commit = GitContext::get_commit(env, None)?;
+                committer = GitContext::get_committer(env, None)?;
+                remote_url = GitContext::get_remote_url(env, None)?;
+            }
+        };
```

Comment:
If you like, you can also just return Ok(Self{...}) in these match arms and don't have to initialise empty variables! And then you don't have to do that Ok(Self) on line 43.

```suggestion
        let repo = Repository::discover(env::current_dir()?);
        match repo {
            Ok(repo) => {
                let head = repo.head().ok();
                Ok(Self {
                    branch: GitContext::get_branch(env, head.as_ref())?,
                    commit: GitContext::get_commit(env, head.as_ref())?,
                    committer: GitContext::get_committer(env, head.as_ref())?,
                    remote_url: GitContext::get_remote_url(env, Some(&repo))?,
                    message: None,
                })
            }
            Err(_) => Ok(Self {
                branch: GitContext::get_branch(env, None)?,
                commit: GitContext::get_commit(env, None)?,
                committer: GitContext::get_committer(env, None)?,
                remote_url: GitContext::get_remote_url(env, None)?,
                message: None,
            }),
        }
```